src/state_machine.zig

```diff
@@ -594,6 +745,653 @@ test "linked accounts" {
+    //1 phase commit [accepted]
+    try testing.expectEqual(@as(u64, 20), state_machine.get_account(7).?.*.debits_accepted);
+    try testing.expectEqual(@as(u64, 0), state_machine.get_account(7).?.*.credits_accepted);
+    try testing.expectEqual(@as(u64, 20), state_machine.get_account(8).?.*.credits_accepted);
+    try testing.expectEqual(@as(u64, 0), state_machine.get_account(8).?.*.debits_accepted);
+
+    //rollback transfer with id [12], amount of 10...
+    state_machine.create_transfer_rollback(state_machine.get_transfer(tran_test_vector[11].input.id).?.*);
+    try testing.expectEqual(@as(u64, 10), state_machine.get_account(7).?.*.debits_accepted);
+    try testing.expectEqual(@as(u64, 0), state_machine.get_account(7).?.*.credits_accepted);
+    try testing.expectEqual(@as(u64, 10), state_machine.get_account(8).?.*.credits_accepted);
+    try testing.expectEqual(@as(u64, 0), state_machine.get_account(8).?.*.debits_accepted);
+    try testing.expect(state_machine.get_transfer(tran_test_vector[11].input.id) == null);
+
+    //rollback transfer with id [15], amount of 10...
+    state_machine.create_transfer_rollback(state_machine.get_transfer(tran_test_vector[22].input.id).?.*);
+    try testing.expectEqual(@as(u64, 0), state_machine.get_account(7).?.*.debits_reserved);
+    try testing.expectEqual(@as(u64, 0), state_machine.get_account(8).?.*.credits_reserved);
+    try testing.expect(state_machine.get_transfer(tran_test_vector[22].input.id) == null);
+}
+
+test "create/lookup/rollback commits" {
+    var arena = std.heap.ArenaAllocator.init(std.heap.page_allocator);
+    defer arena.deinit();
+    const allocator = &arena.allocator;
+
+    const CommitTestVector = struct { outcome: CommitTransferResult, input: Commit };
+
+    var accounts = [_]Account{
+        std.mem.zeroInit(Account, .{ .id = 1 }),
+        std.mem.zeroInit(Account, .{ .id = 2 }),
+        std.mem.zeroInit(Account, .{ .id = 3 }),
+        std.mem.zeroInit(Account, .{ .id = 4 }),
+    };
+
+    var transfers = [_]Transfer{
+        std.mem.zeroInit(Transfer, .{
+            .id = 1,
+            .amount = 15,
+            .debit_account_id = 1,
+            .credit_account_id = 2,
+        }),
+        std.mem.zeroInit(Transfer, .{
+            .id = 2,
+            .amount = 15,
+            .debit_account_id = 1,
+            .credit_account_id = 2,
+            .flags = .{ .two_phase_commit = true },
+            .timeout = 25,
+        }),
+        std.mem.zeroInit(Transfer, .{
+            .id = 3,
+            .amount = 15,
+            .debit_account_id = 1,
+            .credit_account_id = 2,
+            .flags = .{ .two_phase_commit = true },
+            .timeout = 25,
+        }),
+        std.mem.zeroInit(Transfer, .{
+            .id = 4,
+            .amount = 15,
+            .debit_account_id = 1,
+            .credit_account_id = 2,
+            .flags = .{ .two_phase_commit = true },
+            .timeout = 1,
```

Comment:
Curious if there is a particular reason we use a timeout of 1 here instead of 25 as elsewhere?